trino/client.py

```diff
@@ -686,6 +686,10 @@ def execute(self, additional_http_headers=None) -> TrinoResult:
         if self.cancelled:
             raise exceptions.TrinoUserError("Query has been cancelled", self.query_id)
 
+        if self._experimental_python_types:
+            new_header = {constants.HEADER_CLIENT_CAPABILITIES: 'PARAMETRIC_DATETIME'}
+            additional_http_headers = new_header | additional_http_headers if additional_http_headers else new_header
```

Comment:
avoid using dict unions since they overwrite any existing values which can lead to subtle bugs being introduced.

Ideally we'd want to complain if a header whose value was already set in some-place is then tried to be overwritten (either because of bug in our code or because user is trying to access internals of client).

Reply:
I see you changed logic in later commits.